doc/03_research/03_planning/07_reevaluation_desicion_making.md

```diff
@@ -0,0 +1,86 @@
+# Re-evaluation of decision making component
+
+**Summary:** This page gives a foundation for the re-evaluation of the decision-making
+
+---
+
+## Author
+
+Josef Kircher
+
+## Date
+
+26.01.2023
+
+## Prerequisite
+
+---
+<!-- TOC -->
+* [Re-evaluation of decision making component](#re-evaluation-of-decision-making-component)
+  * [**Summary:** This page gives a foundation for the re-evaluation of the decision-making](#summary-this-page-gives-a-foundation-for-the-re-evaluation-of-the-decision-making)
+  * [Author](#author)
+  * [Date](#date)
+  * [Prerequisite](#prerequisite)
+  * [Reasons for re-evaluation](#reasons-for-re-evaluation)
+  * [Options](#options)
+    * [Pylot](#pylot)
+    * [Pytrees](#pytrees)
+      * [Pros](#pros)
+      * [Cons](#cons)
+  * [Conclusion](#conclusion)
+    * [Sources](#sources)
+<!-- TOC -->
+## Reasons for re-evaluation
+
+In the last sprint, I tried to get a graphic tool to work with the docker container withing the project. That failed, but I still think, that a graphical representation would be helpful.
+Other reasons are:
+
+* not much time has been allocated for the state machine so far
+* using SMACH would result in a mostly from scratch implementation
+* harder to debug due to the lack of a graphic representation
+
+## Options
+
+### Pylot
+
+Uses a simple state machine, mostly working with loggers, the skeleton of the state machine might be insteresting,
+as there a different states for preparing and executing a lane change, but overall it might be more interesting for the local planner as most of the interesting stuff is centered around obstacle evasion and avoidance.
+
+### Pytrees
+
+In the last sprint review, Lennart gave me a hint to look at the solution of a former team that used the decision tree library [pytrees](https://py-trees.readthedocs.io/en/devel/index.html).
+
+It is very similar to a finite state machine, so I further investigated both the code of the previous team and the library with its documentation as well.
+
+As it is looking very promising, I list here a few arguments to help support my decision.
+
+#### Pros
+
+* support a graphical representation at runtime with rqt
+* a lot of similar driving scenarios as the old team
+* so a lot of code can be recycled
+* quite intuitive and easy to understand
+* only a limited amount of commands (easy to learn)
+* well documented
+* maintained
+
+#### Cons
+
+* only a couple of decision can be made inside the tree, so it might be more complicated to depict the complex behaviour of the ego vehicle
```

Comment:
What kind of decisions can be handeled with pytrees? I cannot imagine what that means for complex scenarios

Reply:
https://py-trees.readthedocs.io/en/devel/composites.html

there are selectors which function as some sort of 'if-statement' and sequence which execute children sequentially